### ISSUE TITLE
libgroove depend on libsoundio,which can not be compiled successfully

### ISSUE THREAD

**Issue by yunnysunny** [andrewrk/libgroove#120]
run `cmake` in project libgroove ,it will print:
```
System Dependencies
-------------------
* threads                      : OK
* libsoundio                   : not found
* ebur128                      : OK
* chromaprint                  : OK
* libavformat                  : OK
* libavcodec                   : OK
* libavfilter                  : OK
* libavutil                    : OK
```
I downloaded libsoudio, `cmake` ran ok, but when run `make`,it would print:
```
cc1plus: error: unrecognized command line option '-std=c++11'
cc1plus: error: unrecognized command line option "-Wno-c99-extensions" [-Werror]
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/libsoundio_shared.dir/src/soundio.cpp.o] Error 1
make[1]: *** [CMakeFiles/libsoundio_shared.dir/all] Error 2
make: *** [all] Error 2
```
I update my g++ to version 4.9 (under the guide of [Raspberry Pi - Install GCC 4.9 and compile C++14 programs](https://solarianprogrammer.com/2015/01/13/raspberry-pi-raspbian-install-gcc-compile-cpp-14-programs/) ), but it already reported the error `cc1plus: error: unrecognized command line option "-Wno-c99-extensions" [-Werror]`, i deleted the cflag `-Wno-c99-extensions` in the `CMakeLists.txt`, at the end, it resulted in errors as follows:
```
[  2%] Building CXX object CMakeFiles/libsoundio_shared.dir/src/soundio.cpp.o
In file included from /home/pi/libsoundio/src/alsa.hpp:13:0,
                 from /home/pi/libsoundio/src/soundio.hpp:23,
                 from /home/pi/libsoundio/src/soundio.cpp:8:
/home/pi/libsoundio/src/atomics.hpp:27:2: error: #error "require atomic_bool to be lock free"
 #error "require atomic_bool to be lock free"
  ^
In file included from /home/pi/libsoundio/src/soundio.hpp:23:0,
                 from /home/pi/libsoundio/src/soundio.cpp:8:
/home/pi/libsoundio/src/alsa.hpp:48:5: error: 'snd_pcm_chmap_t' does not name a type
     snd_pcm_chmap_t *chmap;
     ^
/home/pi/libsoundio/src/alsa.hpp:68:5: error: 'snd_pcm_chmap_t' does not name a type
     snd_pcm_chmap_t *chmap;
     ^
make[2]: *** [CMakeFiles/libsoundio_shared.dir/src/soundio.cpp.o] Error 1
make[1]: *** [CMakeFiles/libsoundio_shared.dir/all] Error 2
make: *** [all] Error 2
```
I have read course of [Turn Your Raspberry Pi Into a Music Player Server](http://andrewkelley.me/post/raspberry-pi-music-player-server.html#install-groove-basin),i'sure something is different form the condition i have met.

**Comment by andrewrk** [comment]
It looks like you're trying to use master branch, which I don't recommend at this time. I need to fix some important bugs before releasing 5.0.0.

That said, it should at least *build* for you.

I have a raspberry pi and I will use it to test this. Thank you for the bug report.

**Comment by andrewrk** [comment]
I'm pretty sure this means your OS is too old. libgroove requires linux 3.7+.

I did test libsoundio on an rpi and it compiled and ran fine.
